### PR TITLE
fix(account-lib): fix processSigning method of cspr transactionBuilder

### PR DESCRIPTION
**Pull request by CapnMigraine**
Set signatures on txBuilder deploy only once per call to the build() method

Depends on: #1047 

Ticket: STLX-2077

### CODE REVIEW

**Review by rliberal-simtlix — APPROVED** [review]

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts [hunk 1]
```diff
@@ -305,12 +306,35 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
...
   }
 
+  /**
+   * Checks whether the transaction has the owner signature
+   * @param {string} pub - public key of the signer
+   * @returns {boolean} true if the pub key already signed th transaction
+   * @private
+   */
+  private isTransactionSignedByPub(pub: string): boolean {
+    return _.findIndex(this.transaction.casperTx.approvals, approval => {
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/cspr/transactionBuilder.ts:316]: Function looks good, but difficult to read.

**Review by rliberal-simtlix — DISMISSED** [review]

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts [hunk 2]
```diff
@@ -305,12 +306,35 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
...
+      const approvalSigner = approval.signer.slice(2).toUpperCase();
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/cspr/transactionBuilder.ts:317]: Better if you can use something like this.
https://github.com/BitGo/BitGoJS/blob/STLX-2076-cspr-fix-addSignature/modules/account-lib/src/coin/cspr/constants.ts#L1

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts [hunk 3]
```diff
@@ -305,12 +306,35 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
...
+      return approvalSigner === pub;
+    }) === -1;
```
Comment (smaria95) [inline modules/account-lib/src/coin/cspr/transactionBuilder.ts:319]: Shouldn't this be `!== -1`?
For what I can see, findIndex would return a number >= 0 iff `pub` has signed the transaction. So I think it should return true in that case.

**Review by smaria95 — CHANGES_REQUESTED** [review]
Just one correction.

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts (hunk 3 above)
Comment (CapnMigraine) [reply modules/account-lib/src/coin/cspr/transactionBuilder.ts]: You're right, I had the logic inverted and the checks in transactionBuilder.ts too.
Fixed.

File: modules/account-lib/src/coin/cspr/transactionBuilder.ts (hunk 2 above)
Comment (CapnMigraine) [reply modules/account-lib/src/coin/cspr/transactionBuilder.ts:317]: Fixed with a new utils method called `removeAlgoPrefixFromHexValue`

**Review by rliberal-simtlix — APPROVED** [review]

**Review by smaria95 — APPROVED** [review]
LGTM!

**Pull request merged**